### PR TITLE
Specify secret-sharing scheme

### PR DESCRIPTION
**Pull request by alxdavids**
- Defines secret-sharing functions
- Do we need some extra explaining of finite field choice, and also how operations are performed?
- Requires specifying implementation of how reports are constructed using secret sharing scheme

### CODE REVIEW

File: draft-dss-star.md [hunk 1]
```diff
@@ -171,17 +178,50 @@ A threshold secret sharing scheme with the following important properties:
...
   shares correspond to the same underlying secret. Otherwise, it fails.
 
-A threshold secret sharing scheme with these properties has the following syntax:
+A threshold secret sharing scheme with these properties has the following API syntax:
 
-- Share(k, n, msg, rand, tag): Produce a k-out-of-n share of the secret `msg` bound to
-  the auxiliary data `tag` using randomness `rand`. The values k and n are integers,
-  and `msg`, `rand`, and `tag` are byte strings.
-- Recover(share_set): Combine the secret shares in `share_set`, which is of size at
+- Share(k, n, msg, rand, tag): Produce a k-out-of-n share of the secret
```
Comment (chris-wood) [inline draft-dss-star.md:183]:
```suggestion
- Share(k, msg, rand): Produce a k-threshold share of the secret
```
This signature doesn't match the function description below. I also don't know what the best way to describe this "thing" is, as we don't need to describe the number of points at which to evaluate the polynomial (n). I went with "k-threshold share," but there's probably something better.

File: draft-dss-star.md [hunk 2]
```diff
@@ -171,17 +178,50 @@ A threshold secret sharing scheme with the following important properties:
...
+  `x` using randomness `rand`. The values k and n are integers,
+  and `x` and `rand` are byte strings.
```
Comment (chris-wood) [inline draft-dss-star.md:185]:
```suggestion
  `x` using randomness `rand`. The value k is an integer, and `msg` and `rand` are byte strings.
```

File: draft-dss-star.md [hunk 3]
```diff
@@ -171,17 +178,50 @@ A threshold secret sharing scheme with the following important properties:
...
+- Recover(k, n, share_set): Combine the secret shares in `share_set`, which is of size at
```
Comment (chris-wood) [inline draft-dss-star.md:186]:
```suggestion
- Recover(k, share_set): Combine the secret shares in `share_set`, which is of size at
```

File: draft-dss-star.md [hunk 4]
```diff
@@ -171,17 +178,50 @@ A threshold secret sharing scheme with the following important properties:
...
   least k, and recover the corresponding message `msg`. If recovery fails, this function
   returns an error.
 - Nshare: The size in bytes of a secret share value.
 
-[[OPEN ISSUE: specify adept secret sharing here]]
+We use traditional Shamir secret sharing (SSS) {{Shamir}} for
+implementing the sharing scheme. This functionality is implemented using
+a finite field `FFp = GF(p)`, where the order `p` is a large enough
+power-of-two or prime (e.g. of length greater than 32 bits). Note that
+(SSS) is unconditionally secure, and thus the size of the field is not
```
Comment (chris-wood) [inline draft-dss-star.md:195]:
```suggestion
SSS is unconditionally secure, and thus the size of the field is not
```

File: draft-dss-star.md [hunk 5]
```diff
@@ -171,17 +178,50 @@ A threshold secret sharing scheme with the following important properties:
...
+important from a security perspective.
+
+[[OPEN ISSUE: is this size field sufficient for all applications]]
```
Comment (chris-wood) [inline draft-dss-star.md:198]: We need to actually pick a prime to make this interoperable.

File: draft-dss-star.md [hunk 6]
```diff
@@ -171,17 +178,50 @@ A threshold secret sharing scheme with the following important properties:
...
+  polynomial of degree `N-1` from the set `points` of size `N`.
+
+~~~~~
+def Share(k, x, rand):
+  poly = [hash_to_field(x, 1)]
+  poly.extend(hash_to_field(rand, k-1))
+  r = FFp.random()
+  share = 0
+  for i in 0..k
+    share += poly[i] * (rand^i)
```
Comment (chris-wood) [inline draft-dss-star.md:216]: If we're going to use `polynomial_interpolation` from FROST, can we also just use `polynomial_evaluate` to evaluate the polynomial over a random point (`r`)?

**Review by chris-wood — CHANGES_REQUESTED** [review]

File: draft-dss-star.md (hunk 1 above)
Comment (alxdavids) [reply draft-dss-star.md:183]: `k-threshold` seems fine to me

**Review by chris-wood — APPROVED** [review]

**Pull request merged**